pkg/gardenlet/controller/seed/seed/components.go

```diff
@@ -554,13 +554,30 @@ func (r *Reconciler) newCachePrometheus(log logr.Logger, seed *seedpkg.Seed, see
 		return nil, fmt.Errorf("failed getting additional scrape configs: %w", err)
 	}
 
+	var (
+		volumeSize       = "10Gi"
+		retentionSize    = "5GB"
+		storageClassName *string
+	)
+
+	if monitoring := r.Config.Monitoring; monitoring != nil && monitoring.Seed != nil {
+		if storage := monitoring.Seed.PrometheusCache.Storage; storage != nil && storage.Capacity != nil {
+			volumeSize = storage.Capacity.String()
+			storageClassName = storage.ClassName
+		}
+		if !monitoring.Seed.PrometheusCache.Retention.IsZero() {
+			retentionSize = monitoring.Seed.PrometheusCache.Retention.String()
+		}
```

Comment:
This is the same three times. Does it make sense to write a function for this part (taking a `PrometheusConfig` and returning storage class name and retention size)?